lib/jbuilder/jbuilder_template.rb

```diff
@@ -120,7 +121,7 @@ def array!(collection = [], *args)
 
     if args.one? && _partial_options?(options)
       options[:collection] = collection
-      partial! options
+      _render_partial_with_options options
```

Comment:
For context, this is under `array!`. While I am not trying to optimize `array!` in this iteration, making this change made it a lot easier for me to adjust the `partial!` signature.

This will also make `array!` better because it prevents it from having to go _back into_ `partial!` and do a bunch of extra work all over again (ex: another `one?` call, another round of managing options, etc.).

You see this type of thing all over the place in `jbuilder` where one DSL will call another, which results in a bunch of extra over head. In this case, an extra array would be allocated by simply calling `partial!` because of the splat for `*args`.